org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java

```diff
@@ -92,24 +82,73 @@ public String toString() {
 		return version;
 	}
 
+	/**
+	 * Tries to read the Xtext version from this bundle's manifest. The method opens the
+	 * manifest of org.eclipse.xtext.util and reads the <code>Maven-Version</code> value from
+	 * it. In CI scenarios this will be "unspecified", then it continues to read the
+	 * <code>Bundle-Version</code> value. If this ends with ".qualifier", it replaces the qualifier
+	 * by "-SNAPSHOT". Otherwise it will return the base version without the qualifier.
+	 * 
+	 * @return Version string in format <code>MAJOR.MINOR.MICRO[-SNAPSHOT]</code>
+	 */
 	private static String readVersionFromManifest() {
-		URL baseURL = XtextVersion.Plugin.INSTANCE.getBaseURL();
-		try (InputStream is = new URL (baseURL + "META-INF/MANIFEST.MF").openStream()) {
-			Manifest manifest = new Manifest(is);
-			String version = manifest.getMainAttributes().getValue("Maven-Version");
-			if ("unspecified".equals(version)) {
-				version = manifest.getMainAttributes().getValue("Bundle-Version");
-				if (version.endsWith(".qualifier")) {
-					return version.replace(".qualifier", "-SNAPSHOT");
-				} else {
-					return version.substring(0, version.lastIndexOf("."));
+		try {
+			URL ownManifestURL = getOwnManifestURL();
+			
+			try (InputStream is = ownManifestURL.openStream()) {
+				Manifest manifest = new Manifest(is);
+				String version = manifest.getMainAttributes().getValue("Maven-Version");
```

Comment:
Please add a guard here for the bundle name to ensure, that we are working with the correct manifest

Reply:
Makes sense, will do.